ethcore/src/miner/miner.rs

```diff
@@ -263,6 +263,9 @@ impl Miner {
 				let number = block.block().fields().header.number();
 				let difficulty = *block.block().fields().header.difficulty();
 				sealing_work.push(block);
+				if self.work_poster.is_some() {
```

Comment:
Placed it there to avoid locking the `sealing_work` mutex again. Notification is done with the mutex released